extension/opampextension/config_test.go

```diff
@@ -41,20 +41,141 @@ func TestUnmarshalConfig(t *testing.T) {
+		InstanceUID  string
+		Capabilities Capabilities
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "WS must have endpoint",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS: &commonFields{},
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp server endpoint must be provided", err.Error())
+			},
+		},
+		{
+			name: "WS valid endpoint and invalid instance id",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS: &commonFields{
+						Endpoint: "wss://127.0.0.1:4320/v1/opamp",
+					},
+				},
+				InstanceUID: "01BX5ZZKBKACTAV9WEVGEMMVRZFAIL",
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp instance_uid is invalid", err.Error())
+			},
+		},
+		{
+			name: "WS valid endpoint and valid instance id",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS: &commonFields{
+						Endpoint: "wss://127.0.0.1:4320/v1/opamp",
+					},
+				},
+				InstanceUID: "01BX5ZZKBKACTAV9WEVGEMMVRZ",
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "HTTP must have endpoint",
+			fields: fields{
+				Server: &OpAMPServer{
+					HTTP: &commonFields{},
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp server endpoint must be provided", err.Error())
+			},
+		},
+		{
+			name: "HTTP valid endpoint and invalid instance id",
+			fields: fields{
+				Server: &OpAMPServer{
+					HTTP: &commonFields{
+						Endpoint: "https://127.0.0.1:4320/v1/opamp",
+					},
+				},
+				InstanceUID: "01BX5ZZKBKACTAV9WEVGEMMVRZFAIL",
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp instance_uid is invalid", err.Error())
+			},
 		},
+		{
+			name: "HTTP valid endpoint and valid instance id",
+			fields: fields{
+				Server: &OpAMPServer{
+					HTTP: &commonFields{
+						Endpoint: "https://127.0.0.1:4320/v1/opamp",
+					},
+				},
+				InstanceUID: "01BX5ZZKBKACTAV9WEVGEMMVRZ",
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "neither config set",
+			fields: fields{
+				Server: &OpAMPServer{},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp server must have at least ws or http set", err.Error())
+			},
+		},
+		{
+			name: "both config set",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS:   &commonFields{},
+					HTTP: &commonFields{},
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
+				return assert.Equal(t, "opamp server must have only ws or http set", err.Error())
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := &Config{
+				Server:       tt.fields.Server,
+				InstanceUID:  tt.fields.InstanceUID,
+				Capabilities: tt.fields.Capabilities,
+			}
+			tt.wantErr(t, cfg.Validate(), "Validate()")
```

Comment:
```suggestion
			tt.wantErr(t, cfg.Validate())
```
Will this be used anywhere? Right now I don't think any of the functions make use of it.

Reply:
boilerplate from IDE 🙃